xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -343,76 +369,60 @@ XWiki.index.trash.documents.confirmDelete = function(event, table, rowIdx) {
- * Re-display it when the document loads and Javascript is available.
- */
-document.observe('xwiki:dom:loaded', function() {
-  if ($('documentsTrash')) {
-    $('documentsTrash').up('.hidden').removeClassName('hidden');
-  }
+require(['jquery', 'xwiki-meta', 'xwiki-events-bridge'], function($, xm) {
+  //
+  // Delete All
+  //
+  var deleteDocumentConfirmModal = $('#deleteDocumentConfirmModal');
+  deleteDocumentConfirmModal.on('show.bs.modal').find('.btn-danger').click(function() {
+    var notification = new XWiki.widgets.Notification(
+      $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteInProgress')),
+      'inprogress'
+    );
+    $.post(XWiki.currentDocument.getURL('delete'), {
+        'confirm': 1,
+        'emptybin': true,
+        'form_token': xm.form_token,
+        'ajax': 1
+    }).done(function() {
+      notification.replace(new XWiki.widgets.Notification(
+        $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteDone')),
+        'done'
+      ));
+      trashDocumentsLivetable.refresh();
+    }).fail(function() {
+      notification.replace(new XWiki.widgets.Notification(
+        $jsontool.serialize($services.localization.render('platform.index.trashDocumentsDeleteFailed')),
+        'error'
+      ));
+    });
+  });
+  $(document).on('xwiki:livetable:documentsTrash:displayComplete', function(event, data) {
+    if ($('#trashAllDeleted').length == 0) {
+      var actionAllContainer = $('#documentsTrash td.xwiki-livetable-display-header-filter:last-child');
+      var trashAll = $('&lt;a&gt;').attr({
+        'id': 'trashAllDeleted',
+        'class': 'btn btn-danger',
+        'href':'#',
+        'data-toggle': 'modal',
+        'data-target': '#deleteDocumentConfirmModal'
+      }).text("$services.localization.render('index.deletedDocuments.deleteAll')");
+      trashAll.appendTo(actionAllContainer);
+    }
+    $('#trashAllDeleted').toggleClass('disabled', trashDocumentsLivetable.totalRows &lt;= 0);
+  });
 });
 
-// we need to wait the complete display of the livetable to get the number of rows
-document.observe('xwiki:livetable:documentsTrash:displayComplete', function() {
-  // avoid to get multiple times the button
-  if (!$('trashAllDeleted')) {
-    var actionAllContainer = $('documentsTrash').
-      getElementsBySelector('td.xwiki-livetable-display-header-filter:last-child')[0];
-
-    var trashAll = new Element('a', {
-      'id': 'trashAllDeleted',
-      'class': 'btn btn-danger',
-      'href':'#'
-    }).update("$services.localization.render('index.deletedDocuments.deleteAll')");
-    actionAllContainer.insert(trashAll);
-    trashAll.observe('click', XWiki.index.trash.documents.confirmDeleteAll.bindAsEventListener(trashAll));
+/*
+* The livetable is hidden when javascript is disabled, so that an alternative &lt;noscript&gt; table is displayed.
+* Re-display it when the document loads and Javascript is available.
```

Comment:
I don't understand. AFAIK when you enable / disable the JavaScript code the browser reloads the entire page so I don't see why you need to do something special about it.

Reply:
OK.. I haven't seen this done in any other places. Handling no-JavaScript case is definitely not a priority. If we follow this then whenever we use a live table we need to write some JavaScript code to un-hide it, and we use the live table in lots of places. If we really want to support no-JavaScript it should be handled in ``livetable.js`` (but I don't think it is worth). Anyway, it's not that important.